python/tvm/target/datatype.py

```diff
@@ -135,8 +166,40 @@ def lower(op):
             dtype = "uint" + str(t.bits)
             if t.lanes > 1:
                 dtype += "x" + str(t.lanes)
-        if isinstance(op, (_Cast, _FloatImm)):
-            return tvm.tir.call_pure_extern(dtype, extern_func_name, op.value)
-        return tvm.tir.call_pure_extern(dtype, extern_func_name, op.a, op.b)
+        if isinstance(op, _Cast):
+            src_bits = bit_length(op.value.dtype)
+            return call_pure_extern(dtype, extern_func_map[(src_bits, t.bits)], op.value)
+        if isinstance(op, _FloatImm):
+            return call_pure_extern(dtype, extern_func_map[t.bits], op.value)
+        if isinstance(op, _Call):
+            return call_pure_extern(dtype, extern_func_map[t.bits], *op.args)
+        if isinstance(op, _BinaryOpExpr):
+            return call_pure_extern(dtype, extern_func_map[t.bits], op.a, op.b)
```

Comment:
@gussmith23 should we improve debugging message here?

if the map does not contain the bit_width, this throws a KeyError on `extern_func_map[t.bits]`, which may be somewhat cryptic to the user

Reply:
Yeah, please do! Better error messages are never a bad idea 😄